### DISCUSSION TITLE
Combine  FinalClass and HideUtilityClassConstructor behavior

### DISCUSSION THREAD

**Discussion by Dragas** [checkstyle/checkstyle#14905] [General]
I've come into an issue where checkstyle detects that a class is a utility class using `HideUtilityClassConstructor` rule which asserts if a class has only static members. Hiding the said constructor with the private modifier starts triggering the `FinalClass` rule which requires that classes that have only private constructors be final. On paper this sounds okay, but in reality the `FinalClass` rule violation is hidden when a class violates `HideUtilityClassConstructor` rule. Is there a way to show both of these violations at the same time, or perhaps a rule that combines the two behaviors?

**Comment by nrmancuso** [comment]
Please share example code and the configuration that you are using, similar to our bug report template.

**Comment by Dragas** [comment]
```bash
/var/tmp $ javac YOUR_FILE.java
#[[MAKE SURE THERE IS SUCCESSFUL COMPILATION]]

/var/tmp $ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        http://www.puppycrawl.com/dtds/configuration_1_3.dtd>
<module name="Checker">
    <property name="fileExtensions" value="java"/>
    <module name="TreeWalker">
        <module name="FinalClass"/>
        <module name="HideUtilityClassConstructor"/>
    </module>
</module>


/var/tmp $ cat Utility.java

public class Utility {
    public static final String foo = "bar";
    public static String createString() {
        return "";
    }
}

/var/tmp $ cat PrivateConstructorUtility.java

public class PrivateConstructorUtility {
    private PrivateConstructorUtility() {

    }
    public static String  bar(String[] args) {
        return args[0];
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-9.3-all.jar -c config.xml *.java
[ERROR] /var/tmp/PrivateConstructorUtility.java:3:1: Class PrivateConstructorUtility should be declared as final. [FinalClass]
[ERROR] /var/tmp/Utility.java:3:1: Utility classes should not have a public or default constructor. [HideUtilityClassConstructor]
Audit done.
Checkstyle ends with 2 errors.
```
The utility class does not trigger `FinalClass` rule as expected until you add the private constructor to resolve `HideUtilityClassConstructor` violation (see private utility class example).
Individually the rules work fine, and this is expected behavior. I am aware they cannot gather intent from incomplete information, hence why im looking for some combination rule.
Note: gradle has 9.3 as default version
